nagios/status.go

```diff
@@ -0,0 +1,631 @@
+		options = append(options, "unreachable")
+	}
+	if h.Pending {
+		options = append(options, "pending")
+	}
+
+	return buildOptions(options)
+}
+
+type GeneralHostRequest struct {
+	FormatOptions                  FormatOptions
+	Start                          int
+	Count                          int
+	ParentHost                     string
+	ChildHost                      string
+	ShowDetails                    bool
+	DateFormat                     string
+	HostGroup                      string
+	HostStatus                     HostStatus
+	ContactGroup                   string
+	CheckTimeperiodName            string
+	HostNotificationTimeperiodName string
+	CheckCommandName               string
+	EventHandlerName               string
+	ContactName                    string
+	HostTimeField                  string
+	StartTime                      int64
+	EndTime                        int64
+}
+
+func (g GeneralHostRequest) build(query string, includeStartCount bool) Query {
+	q := Query{
+		Endpoint: statusEndpoint,
+		URLQuery: make(url.Values),
+	}
+
+	q.SetNonEmpty("query", query)
+	q.SetNonEmpty("formatoptions", g.FormatOptions.String())
+
+	if includeStartCount {
+		q.URLQuery.Set("start", strconv.Itoa(g.Start))
+		if g.Count > 0 {
+			q.URLQuery.Set("count", strconv.Itoa(g.Count))
+		}
+	}
+
+	q.SetNonEmpty("parenthost", g.ParentHost)
+	q.SetNonEmpty("childhost", g.ChildHost)
+
+	if g.ShowDetails {
+		q.URLQuery.Set("details", strconv.FormatBool(g.ShowDetails))
+	}
+
+	q.SetNonEmpty("dateformat", g.DateFormat)
+	q.SetNonEmpty("hostgroup", g.HostGroup)
+	q.SetNonEmpty("hoststatus", g.HostStatus.String())
+	q.SetNonEmpty("contactgroup", g.ContactGroup)
+	q.SetNonEmpty("checktimeperiod", g.CheckTimeperiodName)
+	q.SetNonEmpty("hostnotificationtimeperiod", g.HostNotificationTimeperiodName)
+	q.SetNonEmpty("checkcommand", g.CheckCommandName)
+	q.SetNonEmpty("eventhandler", g.EventHandlerName)
+	q.SetNonEmpty("contactname", g.ContactName)
+
+	q.SetNonEmpty("hosttimefield", g.HostTimeField)
+	q.SetNonEmpty(startTimeKey, strconv.FormatInt(g.StartTime, 10))
+	q.SetNonEmpty(endTimeKey, strconv.FormatInt(g.EndTime, 10))
+
+	return q
+}
+
+type HostCountRequest struct {
+	GeneralHostRequest
+}
+
+func (h HostCountRequest) Build() Query {
+	return h.build("hostcount", false)
+}
+
+type HostListRequest struct {
+	GeneralHostRequest
+}
+
+func (h HostListRequest) Build() Query {
+	return h.build("hostlist", true)
+}
+
+type HostStatusCount struct {
+	Up          int `json:"up"`
+	Down        int `json:"down"`
+	Unreachable int `json:"unreachable"`
+	Pending     int `json:"pending"`
+}
+
+type HostCountData struct {
+	Selectors map[string]json.RawMessage `json:"selectors"`
+	Count     HostStatusCount            `json:"count"`
+}
+
+type HostCount struct {
+	FormatVersion int           `json:"format_version"`
+	Result        Result        `json:"result"`
+	Data          HostCountData `json:"data"`
+}
+
+//TODO(DanielSz50): Handle showDetails check properly
```

Comment:
```suggestion
// TODO(DanielSz50): Handle showDetails check properly.
```

Oh, yeah, we should have that linter enabled.